iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs

```diff
@@ -10,7 +10,7 @@
 
 namespace Microsoft.Azure.Devices.Client.Transport.Amqp
 {
-    internal class AmqpConnectionHolder : IAmqpConnectionHolder, IAmqpUnitManager
+    internal sealed class AmqpConnectionHolder : IAmqpConnectionHolder, IAmqpUnitManager
```

Comment:
No reason to implement the special dispose pattern if this class doesn't need to be inherited from. Express that with the sealed keyword.